src/lib/components/ResFormClassroom.svelte

```diff
@@ -1,9 +1,10 @@
-<script>
+<script lang="ts">
 	import ResFormPool from '$lib/components/ResFormPool.svelte';
+	import type { ReservationData } from '$types';
 
-	export let rsv = null;
+	export let rsv: ReservationData;
 	export let category = 'classroom';
-	export let date;
+	export let date: Date | null = null;
 	export let dateFn = null;
```

Comment:
@nrxszvo i need to be able to define a consistent shape for `dateFn`:
example:
```
const sampleDateFunction = (arg: Date) : string => {
  return Date.toString();
}

dateFn: (d: Date) => string;
```

dateFn type is `(d: Date) => string`